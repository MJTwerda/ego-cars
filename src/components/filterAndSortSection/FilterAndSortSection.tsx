"use client";
import { useState, useEffect, useCallback } from "react";
import { VehicleModelI } from "../vehicleModels/interfaces/models.interfaces";
import { SortOptions } from "./sort.interfaces";
import ModelCard from '../vehicleModels/modelCard/ModelCard';
import FilterSectionChips from "./FilterSectionChips";
import CommonMenu from "../menu/Menu";
import { MenuItemI } from "../menu/menu.interfaces";
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
import ExpandLessIcon from '@mui/icons-material/ExpandLess';

type FilterAndSortProps = { modelList: VehicleModelI[] };

export default function FilterAndSortSection({ modelList }: FilterAndSortProps) {
  const [filteredModels, setFilteredModels] = useState<VehicleModelI[]>(modelList);
  const [filterValue, setFilterValue] = useState("");
  const [sortValue, setSortValue] = useState<SortOptions>(SortOptions.RANDOM);

  const showDataFiltered = useCallback(() => {
    let updatedModels = [...modelList];

    // Filtrado
    if (filterValue) {
      switch (filterValue) {
        case "all":
          updatedModels = modelList;
          break;
        default:
          updatedModels = updatedModels.filter((m) => m.segment.toLowerCase().includes(filterValue.toLowerCase()));
          break;
      }
    }

    // Ordenamiento
    switch (sortValue) {
      case SortOptions.PRICE_ASC:
        updatedModels.sort((a, b) => a.price - b.price);
        break;
      case SortOptions.PRICE_DESC:
        updatedModels.sort((a, b) => b.price - a.price);
        break;
      case SortOptions.OLDEST:
        updatedModels.sort((a, b) => a.year - b.year);
        break;
      case SortOptions.NEWEST:
        updatedModels.sort((a, b) => b.year - a.year);
        break;
      default:
        break;
    }

    setFilteredModels(updatedModels);
  }, [filterValue, sortValue, modelList]);

  useEffect(() => {
    showDataFiltered();
  }, [showDataFiltered]);

  const handleFilterModels = (tabValue: string) => {
    setFilterValue(tabValue);
  };

  const handleSortModels = (sortValue: SortOptions) => {
    setSortValue(sortValue);
  };

  const vehicleSortMenu: MenuItemI[] = [
    {
      label: "Nada",
      value: SortOptions.RANDOM,
      handleClick: () => handleSortModels(SortOptions.RANDOM),
    },
    {
      label: "De menor a mayor precio",
      value: SortOptions.PRICE_ASC,
      handleClick: () => handleSortModels(SortOptions.PRICE_ASC),
    },
    {
      label: "De mayor a menor precio",
      value: SortOptions.PRICE_DESC,
      handleClick: () => handleSortModels(SortOptions.PRICE_DESC),
    },
    {
      label: "Más nuevos primero",
      value: SortOptions.NEWEST,
      handleClick: () => handleSortModels(SortOptions.NEWEST),
    },
    {
      label: "Más viejos primero",
      value: SortOptions.OLDEST,
      handleClick: () => handleSortModels(SortOptions.OLDEST),
    },
  ];

  return (
    <div>
      <div className="flex justify-between mb-11 pb-2 border-b-2 sm:mb-14">
        <FilterSectionChips 
          filterLabel={{ label: "Filtrar por", isStrong: true }} 
          handleFilterModels={handleFilterModels} 
        />
        
        <CommonMenu 
          menuLabel={{ label: "Ordenar por", isStrong: true }}
          menuIconClosed={<ExpandMoreIcon />}
          menuIconOpened={<ExpandLessIcon />}
          menuItems={vehicleSortMenu} 
          justifyItems='start' 
        />
      </div>

      <div className="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-4 gap-x-4 p-4">
        {filteredModels.map((model: VehicleModelI) => (
          <ModelCard key={model.id} vehicleData={model} />
        ))}
      </div>
    </div>
  );
}