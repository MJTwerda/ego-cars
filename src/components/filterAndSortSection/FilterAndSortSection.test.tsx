import { act, fireEvent, render, RenderResult, waitFor } from "@testing-library/react";
import { useRouter } from "next/navigation";
import FilterAndSortSection from "./FilterAndSortSection";
import { FilterAndSortProps } from './FilterAndSortSection';

const mockProps: FilterAndSortProps = {
  modelList: [
    {
      id: 12345,
      segment: 'Segment 1',
      year: 2022,
      price: 10000,
      thumbnail: '/thumbnail1.jpg',
      name: 'Model 1',
      photo: '/img1.jpg',
    },
    {
      id: 67890,
      segment: 'Segment 2',
      year: 2021,
      price: 8000,
      thumbnail: '/thumbnail2.jpg',
      name: 'Model 2',
      photo: '/img2.jpg',
    }
  ]
};

jest.mock('next/navigation', () => ({
  useRouter: jest.fn(),
}));

const mockPush = jest.fn();
(useRouter as jest.Mock).mockReturnValue({ push: mockPush });

describe('FilterAndSortSection', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render', async () => {
    let component: RenderResult;
    await act(async () => {
      component = render(<FilterAndSortSection modelList={mockProps.modelList} />);
    });

    await waitFor(() => {
      expect(component).toMatchSnapshot();
    });
  });

  it('clicking on filter option, the handleFilterModels function should be called', async() => {
    let component: RenderResult;
    await act(async () => {
      component = render(<FilterAndSortSection modelList={mockProps.modelList} />);
    });

    await waitFor(() => {
      const optionCars = component.getByText('Autos');
      expect(optionCars).toBeInTheDocument();
      fireEvent.click(optionCars);

      const optionAll = component.getByText('Todos');
      expect(optionAll).toBeInTheDocument();
      fireEvent.click(optionAll);
    });
  });

  it('clicking on sort options, the handleSortModels function should be called', async () => {
    let component: RenderResult;
    await act(async () => {
      component = render(<FilterAndSortSection modelList={mockProps.modelList} />);
    });

    await waitFor(() => {
      fireEvent.click(component.getAllByTestId('common_menu_icon_btn')[1]); // Por el primer elemento es el del Header

      // M치s viejos primero
      const oldestBtn = component.getByText('M치s viejos primero');
      expect(oldestBtn).toBeInTheDocument();
      fireEvent.click(oldestBtn);

      // M치s nuevos primero
      const newestBtn = component.getByText('M치s nuevos primero');
      expect(newestBtn).toBeInTheDocument();
      fireEvent.click(newestBtn);

      // Menor precio primero
      const priceAsc = component.getByText('De menor a mayor precio');
      expect(priceAsc).toBeInTheDocument();
      fireEvent.click(priceAsc);

      // Mayor precio primero
      const PriceDesc = component.getByText('De mayor a menor precio');
      expect(PriceDesc).toBeInTheDocument();
      fireEvent.click(PriceDesc);

      // Nada
      const random = component.getByText('Nada');
      expect(random).toBeInTheDocument();
      fireEvent.click(random);
    });

  });
});