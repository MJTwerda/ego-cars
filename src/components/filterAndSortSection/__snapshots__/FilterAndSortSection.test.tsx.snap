// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterAndSortSection should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="flex justify-between mb-11 pb-2 border-b-2 sm:mb-14"
        >
          <div
            class="flex w-2/4 sm:w-3/4"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 flex self-center css-rizt0-MuiTypography-root"
            >
              <strong>
                Filtrar por
              </strong>
            </p>
            <div
              class="flex lg:hidden"
            >
              <div>
                <div
                  class="flex justify-center"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 flex self-center w-3/4 css-rizt0-MuiTypography-root"
                  />
                  <button
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
                    data-testid="common_menu_icon_btn"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="ExpandMoreIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
            <div
              class="flex self-center hidden lg:flex lg:justify-around lg:ml-4"
            >
              <div
                class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-1ykktcv-MuiButtonBase-root-MuiChip-root"
                data-testid="filter_section_chips_all"
                role="button"
                tabindex="0"
              >
                <span
                  class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
                >
                  Todos
                </span>
              </div>
            </div>
            <div
              class="flex self-center hidden lg:flex lg:justify-around lg:ml-4"
            >
              <div
                class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-1ykktcv-MuiButtonBase-root-MuiChip-root"
                data-testid="filter_section_chips_Sedan"
                role="button"
                tabindex="0"
              >
                <span
                  class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
                >
                  Autos
                </span>
              </div>
            </div>
            <div
              class="flex self-center hidden lg:flex lg:justify-around lg:ml-4"
            >
              <div
                class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-1ykktcv-MuiButtonBase-root-MuiChip-root"
                data-testid="filter_section_chips_Pickups y Comerciales"
                role="button"
                tabindex="0"
              >
                <span
                  class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
                >
                  Pickups y Comerciales
                </span>
              </div>
            </div>
            <div
              class="flex self-center hidden lg:flex lg:justify-around lg:ml-4"
            >
              <div
                class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-1ykktcv-MuiButtonBase-root-MuiChip-root"
                data-testid="filter_section_chips_SUVs"
                role="button"
                tabindex="0"
              >
                <span
                  class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
                >
                  SUVs y Crossovers
                </span>
              </div>
            </div>
          </div>
          <div>
            <div
              class="flex justify-center"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 flex self-center w-3/4 css-rizt0-MuiTypography-root"
              >
                <strong>
                  Ordenar por
                </strong>
              </p>
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
                data-testid="common_menu_icon_btn"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="ExpandMoreIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          class="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-4 gap-x-4 gap-y-2 p-4"
        >
          <div
            class="group hover:scale-105 transition-transform duration-300"
          >
            <div
              class="flex flex-col items-center p-4"
            >
              <div
                class="flex flex-col items-center"
              >
                <h1
                  class="text-2xl font-semibold group-hover:text-[--color-active] min-h-[56px] text-center"
                >
                  Model 1
                </h1>
                <div
                  class="flex space-x-2 text-gray-600"
                >
                  <p>
                    2022
                  </p>
                  <p>
                    |
                  </p>
                  <p>
                    $ 
                    10.000
                  </p>
                </div>
              </div>
              <div
                class="flex justify-center items-center w-full h-[200px]"
              >
                <img
                  alt="Model 1"
                  class="rounded object-contain"
                  data-nimg="1"
                  decoding="async"
                  height="200"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg1.jpg&w=640&q=75"
                  srcset="/_next/image?url=%2Fimg1.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg1.jpg&w=640&q=75 2x"
                  style="color: transparent;"
                  width="300"
                />
              </div>
              <a
                class="bg-black text-white px-4 py-2 rounded-3xl opacity-0 group-hover:opacity-100 transition-opacity duration-300"
                href="/model-sheet?id=12345"
              >
                Ver Modelo
              </a>
            </div>
          </div>
          <div
            class="group hover:scale-105 transition-transform duration-300"
          >
            <div
              class="flex flex-col items-center p-4"
            >
              <div
                class="flex flex-col items-center"
              >
                <h1
                  class="text-2xl font-semibold group-hover:text-[--color-active] min-h-[56px] text-center"
                >
                  Model 2
                </h1>
                <div
                  class="flex space-x-2 text-gray-600"
                >
                  <p>
                    2021
                  </p>
                  <p>
                    |
                  </p>
                  <p>
                    $ 
                    8.000
                  </p>
                </div>
              </div>
              <div
                class="flex justify-center items-center w-full h-[200px]"
              >
                <img
                  alt="Model 2"
                  class="rounded object-contain"
                  data-nimg="1"
                  decoding="async"
                  height="200"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg2.jpg&w=640&q=75"
                  srcset="/_next/image?url=%2Fimg2.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg2.jpg&w=640&q=75 2x"
                  style="color: transparent;"
                  width="300"
                />
              </div>
              <a
                class="bg-black text-white px-4 py-2 rounded-3xl opacity-0 group-hover:opacity-100 transition-opacity duration-300"
                href="/model-sheet?id=67890"
              >
                Ver Modelo
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="flex justify-between mb-11 pb-2 border-b-2 sm:mb-14"
      >
        <div
          class="flex w-2/4 sm:w-3/4"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 flex self-center css-rizt0-MuiTypography-root"
          >
            <strong>
              Filtrar por
            </strong>
          </p>
          <div
            class="flex lg:hidden"
          >
            <div>
              <div
                class="flex justify-center"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 flex self-center w-3/4 css-rizt0-MuiTypography-root"
                />
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
                  data-testid="common_menu_icon_btn"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                    data-testid="ExpandMoreIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
          <div
            class="flex self-center hidden lg:flex lg:justify-around lg:ml-4"
          >
            <div
              class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-1ykktcv-MuiButtonBase-root-MuiChip-root"
              data-testid="filter_section_chips_all"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
              >
                Todos
              </span>
            </div>
          </div>
          <div
            class="flex self-center hidden lg:flex lg:justify-around lg:ml-4"
          >
            <div
              class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-1ykktcv-MuiButtonBase-root-MuiChip-root"
              data-testid="filter_section_chips_Sedan"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
              >
                Autos
              </span>
            </div>
          </div>
          <div
            class="flex self-center hidden lg:flex lg:justify-around lg:ml-4"
          >
            <div
              class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-1ykktcv-MuiButtonBase-root-MuiChip-root"
              data-testid="filter_section_chips_Pickups y Comerciales"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
              >
                Pickups y Comerciales
              </span>
            </div>
          </div>
          <div
            class="flex self-center hidden lg:flex lg:justify-around lg:ml-4"
          >
            <div
              class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-1ykktcv-MuiButtonBase-root-MuiChip-root"
              data-testid="filter_section_chips_SUVs"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
              >
                SUVs y Crossovers
              </span>
            </div>
          </div>
        </div>
        <div>
          <div
            class="flex justify-center"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 flex self-center w-3/4 css-rizt0-MuiTypography-root"
            >
              <strong>
                Ordenar por
              </strong>
            </p>
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
              data-testid="common_menu_icon_btn"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="ExpandMoreIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-4 gap-x-4 gap-y-2 p-4"
      >
        <div
          class="group hover:scale-105 transition-transform duration-300"
        >
          <div
            class="flex flex-col items-center p-4"
          >
            <div
              class="flex flex-col items-center"
            >
              <h1
                class="text-2xl font-semibold group-hover:text-[--color-active] min-h-[56px] text-center"
              >
                Model 1
              </h1>
              <div
                class="flex space-x-2 text-gray-600"
              >
                <p>
                  2022
                </p>
                <p>
                  |
                </p>
                <p>
                  $ 
                  10.000
                </p>
              </div>
            </div>
            <div
              class="flex justify-center items-center w-full h-[200px]"
            >
              <img
                alt="Model 1"
                class="rounded object-contain"
                data-nimg="1"
                decoding="async"
                height="200"
                loading="lazy"
                src="/_next/image?url=%2Fimg1.jpg&w=640&q=75"
                srcset="/_next/image?url=%2Fimg1.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg1.jpg&w=640&q=75 2x"
                style="color: transparent;"
                width="300"
              />
            </div>
            <a
              class="bg-black text-white px-4 py-2 rounded-3xl opacity-0 group-hover:opacity-100 transition-opacity duration-300"
              href="/model-sheet?id=12345"
            >
              Ver Modelo
            </a>
          </div>
        </div>
        <div
          class="group hover:scale-105 transition-transform duration-300"
        >
          <div
            class="flex flex-col items-center p-4"
          >
            <div
              class="flex flex-col items-center"
            >
              <h1
                class="text-2xl font-semibold group-hover:text-[--color-active] min-h-[56px] text-center"
              >
                Model 2
              </h1>
              <div
                class="flex space-x-2 text-gray-600"
              >
                <p>
                  2021
                </p>
                <p>
                  |
                </p>
                <p>
                  $ 
                  8.000
                </p>
              </div>
            </div>
            <div
              class="flex justify-center items-center w-full h-[200px]"
            >
              <img
                alt="Model 2"
                class="rounded object-contain"
                data-nimg="1"
                decoding="async"
                height="200"
                loading="lazy"
                src="/_next/image?url=%2Fimg2.jpg&w=640&q=75"
                srcset="/_next/image?url=%2Fimg2.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg2.jpg&w=640&q=75 2x"
                style="color: transparent;"
                width="300"
              />
            </div>
            <a
              class="bg-black text-white px-4 py-2 rounded-3xl opacity-0 group-hover:opacity-100 transition-opacity duration-300"
              href="/model-sheet?id=67890"
            >
              Ver Modelo
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
